test/lib/jdk/test/lib/format/ArrayCodec.java

```diff
@@ -0,0 +1,336 @@
+     */
+    public void appendFormatted() {
+        if (exhausted) {
+            return;
+        }
+
+        boolean isLast = idx == source.size() - 1;
+        if (isLast || source.isEmpty()) {
+            exhausted = true;
+        }
+
+        if (bounded && encoded.length() + element.length() > maxWidth - ELLIPSIS.length()) {
+            encoded.append(isLast ? element : " " + ELLIPSIS);
+            exhausted = true;
+        } else {
+            encoded.append(element);
+        }
+        idx++;
+    }
+
+    /**
+     * Aligns the element by another codec.
+     *
+     * If another codec's last encoded element string is longer than this
+     * codec's, widens this codec's encoded element with spaces so the
+     * two strings have the same length;
+     *
+     * @param another Another codec to compare encoded element width with
+     */
+    public void alignBy(ArrayCodec<E> another) {
+        if (!element.equals("") && !element.equals("]")) {
+            int delta = another.element.length() - element.length();
+            if (delta > 0) {
+                element = Format.paddingForWidth(delta) + element;
+            }
+        }
+    }
+
+    /**
+     * Indicates if there are no elements left in the source array
+     *
+     * @return {@code true} if there are no elements left, {@code false} otherwise
+     */
+    public boolean isExhausted() {
+        return exhausted;
+    }
+
+    /**
+     * Returns the string encoded-so-far
+     *
+     * @return the string encoded-so-far
+     */
+    public String getEncoded() {
+        return encoded.toString();
+    }
+
+    /**
+     * Returns the length of the string encoded-so-far
+     *
+     * @return the length of the string encoded-so-far
+     */
+    public int getEncodedLength() {
+        return encoded.length();
+    }
+
+    /**
+     * Returns the length of the last encoded element
+     *
+     * @return the length of the last encoded element
+     */
+    public int getElementLength() {
+        return element.length();
+    }
+
+    /**
+     * Finds and returns the first mismatch index in another codec
+     *
+     * @param another a codec mismatch with whom is to be found
+     * @return the first mismatched element's index
+     */
+    public int findMismatchIndex(ArrayCodec<E> another) {
+        for (int result = 0; (source.size() > result) && (another.source.size() > result); result++) {
+            Object first = source.get(result);
+            Object second = another.source.get(result);
+
+            if (first == null || second == null) {
+                if (first == null && second == null) {
+                    continue;   // Both elements are null (i.e. equal)
+                } else {
+                    return result;  // Only one element is null, here's the failure index
+                }
+            }
+
+            if (!first.equals(second)) {
+                return result;
+            }
+
+        }
+
+        return 0;
```

Comment:
shouldn't it be -1?

Reply:
An interesting notice, actually. It should be either `-1` or `result` depending on whether arrays are equal, or one array is a prefix of another. Fixed in the 174148ab41e1c03d46dbde8862865dbaf3138843.